spring-multirabbit-lib/src/main/java/org/springframework/amqp/rabbit/annotation/MultiRabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -4,31 +4,34 @@
 import org.springframework.amqp.core.Binding;
 import org.springframework.amqp.core.Declarable;
 import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
 import org.springframework.amqp.rabbit.core.RabbitAdmin;
-import org.springframework.beans.factory.BeanFactory;
-import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.ApplicationContextAware;
 
 import java.lang.reflect.Method;
 
 /**
- * An extension of {@link RabbitListenerAnnotationBeanPostProcessor} that attaches the processing of beans
- * for Exchanges, Queues, and Bindings after they are created.
+ * An extension of {@link RabbitListenerAnnotationBeanPostProcessor} that attaches the processing of beans for
+ * Exchanges, Queues, and Bindings after they are created.
  * <p>
  * This processing enables each {@link RabbitAdmin} to differentiate which beans corresponds to that specific
- * {@link RabbitAdmin}, preventing the server from being populated with non-expected structures from other
- * servers.
+ * {@link RabbitAdmin}, preventing the server from being populated with non-expected structures from other servers.
+ *
+ * @author Wander Costa
+ * @see RabbitListenerAnnotationBeanPostProcessor
  */
-public class ExtendedRabbitListenerAnnotationBeanPostProcessor
+public final class MultiRabbitListenerAnnotationBeanPostProcessor
         extends RabbitListenerAnnotationBeanPostProcessor
-        implements ApplicationContextAware, BeanFactoryAware {
-
-    private static final String NO_ADMIN_BEAN_ERROR = "Bean '%s' not a RabbitAdmin. Cannot enhance beans with"
-            + "RabbitAdmin.";
+        implements ApplicationContextAware {
 
     private ApplicationContext applicationContext;
-    private BeanFactory beanFactory;
+
+    // TODO Remove this workaround. It ensures the ConnectionFactory is ready when the listeners
```

Comment:
I remember this trick to autowire the connection factory.
But why is it marked as `//TODO remove`?

Reply:
Yeah, you're right.